api/server.go

```diff
@@ -109,26 +120,29 @@ func (s *service) Dump(ctx context.Context, args *task.DumpArgs) (*task.DumpResp
 
 		multipartCheckpointResp, cid, err := store.CreateMultiPartUpload(checkpointFullSize)
 		if err != nil {
+			err := status.Error(codes.Unavailable, "CreateMultiPartUpload failed")
 			return nil, err
 		}
 
 		err = store.StartMultiPartUpload(cid, &multipartCheckpointResp, checkpointPath)
 		if err != nil {
+			err := status.Error(codes.Unavailable, "StartMultiPartUpload failed")
 			return nil, err
 		}
 
 		err = store.CompleteMultiPartUpload(multipartCheckpointResp, cid)
 		if err != nil {
+			err := status.Error(codes.Unavailable, "CompleteMultiPartUpload failed")
 			return nil, err
 		}
 
 		return &task.DumpResp{
-			Error: fmt.Sprintf("Dumped process %d to %s, multipart checkpoint id: %s", args.PID, args.Dir, multipartCheckpointResp.UploadID),
+			Message: fmt.Sprintf("Dumped process %d to %s, multipart checkpoint id: %s", args.PID, args.Dir, multipartCheckpointResp.UploadID),
 		}, nil
 	}
 
 	return &task.DumpResp{
-		Error: fmt.Sprintf("Dumped process %d to %s", args.PID, args.Dir),
+		Message: fmt.Sprintf("Dumped process %d to %s", args.PID, args.Dir),
```

Comment:
I think you should try stuffing this data into the errorcode metadata like in the blog post, instead kind of like this: 

```
desc := "The username must only contain alphanumeric characters"
v := &errdetails.BadRequest_FieldViolation{
    Field: "username",
    Description: desc,
}
st, err := st.WithDetails(v)
```

It'll help us I think avoiding polluting the grpc w/ random messages and also makes error checking on the other side easier

Reply:
I can see where I can use this elsewhere but that return statement is where err is nil and there was a successful dump

Reply:
oh yup, picked a bad example